concepts/05 UI Components/DataGrid/20 Editing/45 Disable Editing/40 Cell.md

```diff
@@ -0,0 +1,110 @@
+    $(function() {
+        $("#dataGrid").dxDataGrid({
+            // ...
+            onEditorPreparing(e) {
+                if (e.dataField === "LastName" && e.parentType === "dataRow") {
+                    e.editorOptions.disabled = e.row.data && e.row.data.FirstName === "";
+                }
+            }
+        });
+    });
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-data-grid 
+        // ...
+        (onEditorPreparing)="onEditorPreparing($event)"
+    >   
+    </dx-data-grid>
+
+    <!-- tab: app.component.ts -->
+    import { DxDataGridModule } from "devextreme-angular";
+    // ...
+    export class AppComponent {
+        onEditorPreparing(e) {
+            if (e.dataField === "LastName" && e.parentType === "dataRow") {
+                e.editorOptions.disabled = e.row.data && e.row.data.FirstName === "";
+            }
+        }
+    }
+    @NgModule({
+        imports: [
+            // ...
+            DxDataGridModule
+        ],
+        // ...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid 
+            @editor-preparing="onEditorPreparing"
+        >
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import {
+        DxDataGrid,
+        // ...
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            // ...
+        },
+        methods: {
+            onEditorPreparing(e) {
+                if (e.dataField === "LastName" && e.parentType === "dataRow") {
+                    e.editorOptions.disabled = e.row.data && e.row.data.FirstName === "";
+                }
+            }
+        }
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React, { useCallback } from "react";
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DataGrid } from 'devextreme-react/data-grid';
+
+    function App() {
+        // ...
+        const onEditorPreparing = useCallback((e) => {
+            if (e.dataField === "LastName" && e.parentType === "dataRow") {
+                e.editorOptions.disabled = e.row.data && e.row.data.FirstName === "";
+            }
+        }, []);
+
+        return (
+            <DataGrid 
+                onEditorPreparing={onEditorPreparing}
+            >
+            </DataGrid>
+        );
+    }
+
+    export default App;
+
+---
+
+Certain edit modes - like Row Edit Mode - activate multiple editors at once. You may still want user input in one of the cells to enable/disable an editor in another cell. Specify the [setCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#setCellValue) callback function for the column where input values affect other cells. Refer to the following topic for implementation details: [Dynamically Change Editor Properties in the Editing State](/Documentation/Guide/UI_Components/DataGrid/How_To/Dynamically_Change_Editor_Properties_in_the_Editing_State/).
```

Comment:
```suggestion
There are edit modes - such as Row Edit Mode - that activate multiple editors at once. You may still want user input in one of the cells to enable/disable an editor in another cell. Specify the [setCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#setCellValue) callback function for the column where input values affect other cells. Refer to the following topic for implementation details: [Dynamically Change Editor Properties in the Editing State](/Documentation/Guide/UI_Components/DataGrid/How_To/Dynamically_Change_Editor_Properties_in_the_Editing_State/).
```
Is there something missing between the first and second sentence here? I feel like there is a step or an idea missing.

Reply:
_What about this:_ There are edit modes - such as Row Edit Mode - that activate multiple editors at once. In this case, you may need  to enable/disable an editor in one cell when user input in another cells.

Reply:
You can also check this example https://codepen.io/serhiipylypchuk1991/pen/JjBXVEZ?editors=0010

Reply:
"You may still want user input in one of the cells to enable/disable an editor in another cell."
changed to
"When these modes are active, you may need to enable/disable an editor in one cell when user input in another cells." 

This change unfortunately didn't clarify things, but rather confused me. Can you please take a moment to rework it? Don't skip steps. In this case, I would rather you wrote a little too much and then I will edit it for you. I appreciate your time. Thanks!

Reply:
In general it works in the following way: For instance, there is a row with 2 cells. When a user press the Edit button, he can edit each cell separately. If the second cell is empty, the first one will be disabled.

Considering this, the text can be as follows:

There are edit modes - such as Row Edit Mode - that activate multiple editors at once. If the required cell is empty, you can enable/disable an editor in another cells. Specify the setCellValue callback function for the column where input values affect other cells. Refer to the following topic for implementation details: Dynamically Change Editor Properties in the Editing State.

Reply:
Serhii, I am having a hard time following your ideas, so I asked Albert for help. Here is what he had to say:
###
What I see based on my understanding of how the example works:
1) I click the Edit button for the "Jim Packard" row, which becomes the focused row (which might be what he/she means by "edit each cell separately".
2) Cells in all other rows become disabled (try clicking "Harv Mudd" or "Todd Hoffman" to see what I mean).
Based on my own understanding of the CodePen demo, Albert asks if what you mean is:
"When a user focuses a row, the user can edit cells in that specific row, but the cells in all other rows are disabled."
###

Reply:
No, it works in another way. Try to follow this steps:
1) Click the **Edit** button of any row. All cells in this row will be available for editing (separately).
2) Edit the second cell (right one) and make it empty. After this, click the **Save** button to save data.
3) Click the **Edit** button of this row once again and try to edit the first cell (left one). This cell will be disabled becouse of the first cell is empty.

Reply:
I think the following explanation may be clearer:
You can dynamically disable an editor based on another editor's value. Handle the [onEditorPreparing](https://file+.vscode-resource.vscode-cdn.net/c:/Repositories/DX_Docs/Docs/Source/22_1/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditorPreparing) event and specify the required condition within its handler. For instance, if the required cell is empty, you can enable/disable an editor in another cells. Refer to the following topic for implementation details: [Dynamically Change Editor Properties in the Editing State](https://file+.vscode-resource.vscode-cdn.net/c:/Repositories/DX_Docs/Docs/Source/22_1/Documentation/Guide/UI_Components/DataGrid/How_To/Dynamically_Change_Editor_Properties_in_the_Editing_State).

Reply:
I'm not sure why the last version mentions **onEditorPreparing**, but doesn't mention **setCellValue** anymore. As far as I understand, **onEditorPreparing** remains a constant throughout this topic. No matter which edit mode is active, you need to handle that event. What changes in certain modes - those that activate multiple editors at once - is that you need to force an update to active editors. That's why you need the **setCellValue** callback. It refreshes the editors and lets you keep the edit mode active at the same time. Right? :-) 

So I would probably write something like this:

You may need an extra implementation step if the control uses an edit mode that activate multiple editors at once (such as the "Row" edit mode). Specify a [setCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#setCellValue) callback for the column whose values specify editor availability in other columns. That callback helps you refresh the editors - disable or enable them as needed - and keep the control in edit mode. Refer to the following topic for additional information: [Dynamically Change Editor Properties in the Editing State](/Documentation/Guide/UI_Components/DataGrid/How_To/Dynamically_Change_Editor_Properties_in_the_Editing_State/).

Reply:
Thank you Vova!
I will let Vova's edit stand then, Serhii, if you are okay with it.

Also, wanted to let you know that the words "another" and "other" mean the same thing, except that another is used with a singular noun and other is used with uncountable and plural nouns. This means that in the case of the paragraph your wrote above, the correct phrasing would be "For instance, if the required cell is empty, you can enable/disable an editor in **other** cells."

Reply:
Thanks! I will take it into account)